### ISSUE TITLE
Git identities

### ISSUE THREAD

**Issue by jacg** [next-exp/IC#337]
When the email address in a commit does not match the address associated with your GitHub account, GitHub sometimes does not recognize the user correctly.

Please ensure that your git identity is set correctly:

    git config --global user.name "Fulano Mengano"
    git config --global user.email fulano@mengano.com

(You can omit the `--global` to set it on a per-repository basis.)

@abotas in particular had a different email address for every machine he ever worked on.

I have fixed some obvious ones that I spotted. This required a rebase of most of `nextic/master`. Consequently, you you will have to fetch the latest master and rebase your local branches on top of it. The only changes in `nextic/master` as a consequence of this (apart from recent merges) are some modified committer emails. Consequently, you should not see any conflicts during the rebase (unless your previous fetch predates recent PR merges).

If you are aware of any incorrect email addresses in your history, please let me know ASAP, and I will fix them. I would like to do this bureaucratic change only once, as rebasing `nextic/master` is something that should be avoided. (Given that no code changes in this rebase, it's not too drastic in this case, but it still might cause some confusion, so I'd rather not do it again after this weekend.)

In summary:

1. Make sure that your git user id is configured properly
2. Tell me if you are aware of incorrect emails in your commit history that you want fixing.
3. Any local branches you have, will have to be rebased (trivially) on top of a newly-fetched `nextic/master`.

**Comment by paolafer** [comment]
Hi! I'm not sure I know how to do 1., and realize about 2.

**Comment by jacg** [comment]
> I'm not sure I know how to do 1.

As shown in the original comment above:

    git config --global user.name "Paola Ferrario"
    git config --global user.email <whatever email address you registered on GitHub>

You will have to do this on every machine on which you develop, and remember to do it when moving to a new machine.

> and realize about 2.

Grepping through the logs for (case insensitive) 'Paola' finds 12 commits, all with the same email address.

**Comment by paolafer** [comment]
Ok, done it. So, now my name should be Paola Ferrario instead of Paola from now on?

**Comment by jacg** [comment]
> So, now my name should be Paola Ferrario instead of Paola from now on?

For all your future commits, yes. As long as you remember to set it again on any new machine.

I won't go back and fix your human-readable name in history, as GitHub dosen't care about that at all, and I really want to avoid rebasing `master`.

**Comment by jacg** [comment]
@jerenner I can see 3 different addresses for you:

1. On a MacBook
2. a CERN visitor on dyndns
3. on nextp12

Clearly your git identity is not configured.

**Comment by jacg** [comment]
@ansimez Yours needs confiuring too: your commits are registered to an email address on your macbook.

**Comment by jacg** [comment]
@jerenner @ansimez Please communicate to me the email address you have registered with GitHub.

Anyone else, If you would like me to fix your human readable names, speak now or forever hold your peace. I will do one final rebase of `nextic/master` sometime this evening, and then I really don't want to do it again!

Locally I have already made the following changes, awaiting to be pushed to `nextic` in one go:

1. Unified @jjgomezcadenas various names
2. Paola -> Paola Ferrario

@jerenner What is your preferred human-readable name? (The one you are going to set in your git config.) Currently you have

+ jerenner
+ josh

**Comment by Aretno** [comment]
Mine is "aretno@gmail.com". I'll do my git config tomorrow using "Ander Simón" as my human-readable name if accent marks are allowed, if not, "Ander Simon". Thanks!

**Comment by jerenner** [comment]
I'll set my user.name as "Josh Renner"
I'll also set my user.email to: josh.e.renner@gmail.com

I had not set these before, and looking at my commits in Magit, it looks like my name was often followed by something that is not a correct email: for example <jrenner@nextp12.ific.uv.es>.  This seems to be the case for all previous commits I had made in my fork of IC.  Really the only recent relevant commit (PR that actually got merged) is this one: https://github.com/nextic/IC/pull/294

Thanks

**Comment by jacg** [comment]
OK, with that I now have a clean and consistent set of identities in my version of `master`.

@jmbenlloch has human-readable name 'jmbenlloch' and @gonzaponte has human-readable name 'Gonzalo'. If you would like these changed, let me know today. Otherwise I'll push what I have, and hopefully we'll be done with rebasing `nextic/master`.

**Comment by gonzaponte** [comment]
I am not sure if this goes in the line of this issue, but when I merge some
PR, I use the name I see in Github as the author. For instance, jacg,
jjgomezcadenas, etc. Insted of the actual names. Am I doing it wrong?

**Comment by jacg** [comment]
I recall that for some people I had to use the name and for some I had to use the userid, in order to get (ma)git to recognize it. I've never really got to the bottom of that.

Given that (in our current rebase --no-ff policy) you create these merge commits for PRs in your local repo, which is not really aware of GitHub, it's not clear to me how the *GitHub* id matters.

Keep doing what you were doing and let's keep an eye on the results, now that we have consistent ids throughout. If something funny happens, we'll think about it.

**Comment by jacg** [comment]
If @jmbenlloch and @gonzaponte are happy with their single-word human-readable names, I can push now and close this issue.

**Comment by gonzaponte** [comment]
I am.

**Comment by jacg** [comment]
I have pushed the history with the updated identities to `nextic/master`.
